Plugins/translate/TranslatePlugin.cs

```diff
@@ -0,0 +1,73 @@
+﻿using BasePlugin;
+using BasePlugin.Interfaces;
+using BasePlugin.Records;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Text.Json;
+
+namespace List
+{
+    public class TranslatePlugin : IPlugin
+    {
+        public const string _Id = "translate";
+        public string Id => _Id;
+        private static readonly string subscriptionKey = Environment.GetEnvironmentVariable("TranslateKey");
+        private static readonly string endpoint = "https://api.cognitive.microsofttranslator.com/";
+        private static readonly string location = "eastus2";
+
+        public void Execute(PluginInput input)
+        {
+            if (input.Message == "")
+            {
+                input.Callbacks.StartSession();
+                input.Callbacks.SendMessage("Type a sentence to translate. Enter 'len' to which language to translate into. Enter 'Exit' to stop");
+            }
+            else if (input.Message.ToLower() == "exit")
+            {
+                input.Callbacks.EndSession();
+                input.Callbacks.SendMessage("List stopped.");
+            }
+            else if (input.Message.ToLower().StartsWith("len"))
+            {
+                var str = input.Message.Substring("len".Length).Trim();
+                input.Callbacks.SavePluginUserData(str);
+
+                input.Callbacks.SendMessage($"the translation language: {str}");
+            }
+            else
+            {
+                translate(input.Message, input).ToString();
```

Comment:
You should wait on a async function, here you should use '.Wait()'

Reply:
It waits inside the function